comments-bundle/src/EventListener/DataContainer/CommentsConditionalFieldsListener.php

```diff
@@ -0,0 +1,176 @@
+            case 'tl_faq_category':
+                self::applyParentFields($table);
+                break;
+
+            case 'tl_news':
+            case 'tl_calendar_events':
+            case 'tl_faq':
+                self::applyChildrenFields($table);
+                break;
+        }
+    }
+
+    private function applyParentFields(string $table): void
+    {
+        $GLOBALS['TL_DCA'][$table]['palettes']['__selector__'][] = 'allowComments';
+        $GLOBALS['TL_DCA'][$table]['subpalettes']['allowComments'] = 'notify,sortOrder,perPage,moderate,bbcode,requireLogin,disableCaptcha';
+
+        $GLOBALS['TL_DCA'][$table]['fields']['allowComments'] = [
+            'filter' => true,
+            'inputType' => 'checkbox',
+            'eval' => ['submitOnChange' => true],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['notify'] = [
+            'inputType' => 'select',
+            'options' => ['notify_admin', 'notify_author', 'notify_both'],
+            'reference' => &$GLOBALS['TL_LANG'][$table],
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => "varchar(32) NOT NULL default 'notify_admin'",
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['sortOrder'] = [
+            'inputType' => 'select',
+            'options' => ['ascending', 'descending'],
+            'reference' => &$GLOBALS['TL_LANG']['MSC'],
+            'eval' => ['tl_class' => 'w50 clr'],
+            'sql' => "varchar(32) NOT NULL default 'ascending'",
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['perPage'] = [
+            'inputType' => 'text',
+            'eval' => ['rgxp' => 'natural', 'tl_class' => 'w50'],
+            'sql' => 'smallint(5) unsigned NOT NULL default 0',
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['moderate'] = [
+            'inputType' => 'checkbox',
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['bbcode'] = [
+            'inputType' => 'checkbox',
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['requireLogin'] = [
+            'inputType' => 'checkbox',
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        $GLOBALS['TL_DCA'][$table]['fields']['disableCaptcha'] = [
+            'inputType' => 'checkbox',
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        switch ($table) {
+            case 'tl_news':
+            case 'tl_calendar_events':
+                PaletteManipulator::create()
+                    ->addLegend('comments_legend', 'protected_legend', PaletteManipulator::POSITION_AFTER, true)
```

Comment:
```suggestion
                    ->addLegend('comments_legend', null, PaletteManipulator::POSITION_APPEND, true)
```

Do we really need to specify the parent legend? Or could we maybe just append to the end and not have to distinguish between the tables? Not sure my code works as is! 😅